core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1338,27 +1337,22 @@ class KafkaApis(val requestChannel: RequestChannel,
   /**
    * Handle an offset fetch request
    */
-  def handleOffsetFetchRequest(request: RequestChannel.Request): Unit = {
+  def handleOffsetFetchRequest(request: RequestChannel.Request): CompletableFuture[Unit] = {
     val version = request.header.apiVersion
     if (version == 0) {
-      // reading offsets from ZK
-      handleOffsetFetchRequestV0(request)
-    } else if (version >= 1 && version <= 7) {
-      // reading offsets from Kafka
-      handleOffsetFetchRequestBetweenV1AndV7(request)
+      handleOffsetFetchRequestFromZookeeper(request)
```

Comment:
i think it can be more idiomatic and cleaner if we rewrite it using match - case 
```
version match {
  case 0: handle...
  case  ...
}
```

Reply:
Is it really more idiomatic? We only have a special case for version 0 so an `if/else` seems fine here.